evennia/contrib/wilderness.py

```diff
@@ -0,0 +1,490 @@
+    wilderness map it is possible to give a "map provider": this is a python
+    object that is smart enough to create the map.
+
+    The default provider, WildernessMapProvider, just creates a grid area that
+    is unlimited in size.
+    This WildernessMapProvider can be subclassed to create more interesting
+    maps and also to customize the room/exit typeclass used.
+
+    There is also no command that allows players to enter the wilderness. This
+    still needs to be added: it can be a command or an exit, depending on your
+    needs.
+
+
+Implementation details:
+
+    When a character moves into the wilderness, they get their own room. If they
+    move, instead of moving the character, the room changes to match the new
+    location.
+    If a character meets another character in the wilderness, then their room
+    merges. When one of the character leaves again, they each get their own
+    separate rooms.
+    Rooms are created as needed. Unneeded rooms are stored away to avoid the
+    overhead cost of creating new rooms again in the future.
+
+"""
+
+from evennia import DefaultRoom, DefaultExit, DefaultScript
+from evennia import create_object, create_script
+from evennia.utils import inherits_from
+
+
+class WildernessRoom(DefaultRoom):
+
+    @property
+    def wilderness(self):
+        """Shortcut property to the wilderness script this room belongs to."""
+        return self.ndb.wildernessscript
+
+    @property
+    def location_name(self):
+        """Shortuct property to the location name."""
+        return self.wilderness.provider.get_location_name(
+            self.ndb.active_location)
+
+    def at_object_receive(self, moved_obj, source_location):
+        """Called after an object has been moved into this object."""
+        if moved_obj.destination and moved_obj.destination == moved_obj.location:
+            # Ignore exits looping back to themselves: those are the regular
+            # n, ne, ... exits.
+            return
+
+        itemlocs = self.wilderness.db.itemlocations
+        if moved_obj in itemlocs:
+            # This object was already in the wilderness. We need to make sure it
+            # goes to the correct room it belongs to.
+            # Otherwise the following issue can come up:
+            # 1) Player 1 and Player 2 share a room
+            # 2) Player 1 disconnects
+            # 3) Player 2 moves around
+            # 4) Player 1 reconnects
+            # Player 1 will end up in player 2's room, which has the wrong
+            # location
+
+            location = itemlocs[moved_obj]
+            # Setting the location to None is important here so that we always
+            # get a "fresh" room
+            moved_obj.location = None
+            self.wilderness.move_obj(moved_obj, location)
+        else:
+            # This object wasn't in the wilderness yet. Let's add it.
+            itemlocs[moved_obj] = self.ndb.active_location
```

Comment:
While I can see later in the code that self.ndb.active_location is set by the wilderness script, this is the first time we see this property appear, so it would help readability to mention where it comes from, or even mention this in the class docstring.